app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -976,6 +986,26 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
         }
     }
 
+    private fun setDefaultBrowserNotificationIfNeeded() {
+        val instanceWorkManager = WorkManager.getInstance(this)
+
+        if (!settings().shouldShowDefaultBrowserNotification()) {
+                // cancel notification work if already default browser
+                instanceWorkManager.cancelUniqueWork(DefaultBrowserNotificationWorker.NOTIFICATION_WORK_NAME)
+            return
+        }
+
+        val notificationWork = OneTimeWorkRequest.Builder(DefaultBrowserNotificationWorker::class.java)
+            .setInitialDelay(DefaultBrowserNotificationWorker.NOTIFICATION_DELAY, TimeUnit.MILLISECONDS)
```

Comment:
Ah, I think I can answer my own question, but you can confirm: this wouldn't happen because `defaultBrowserNotificationDisplayed` needs to be set to true and that only happens in the processor.

This was a bit confusing to follow because of the (nested) double negative. 😄